Sming/SmingCore/Network/HttpServer.h

```diff
@@ -66,32 +59,51 @@ class HttpServer : public TcpServer
 	/**
 	 * @briefs Allows content-type specific parsing of the body based on content-type.
 	 *
-	 * @param const String& contentType. Can be full content-type like 'application/json', or 'application/*'  or '*'.
-	 * 						If there is exact match for the content-type wildcard content-types will not be used.
-	 * 						There can be only one catch-all '*' body parser and that will be the last registered
+	 * @param contentType. Can be full content-type like 'application/json', or 'application/*'  or '*'.
+	 * 			If there is exact match for the content-type wildcard content-types will not be used.
+	 * 			There can be only one catch-all '*' body parser and that will be the last registered
 	 *
-	 * @param  HttpBodyParserDelegate parser
+	 * @param  parser
 	 */
-	void setBodyParser(const String& contentType, HttpBodyParserDelegate parser);
+	void setBodyParser(const String& contentType, HttpBodyParserDelegate parser)
+	{
+		bodyParsers[contentType] = parser;
+	}
 
 	/**
-	 * @param String path URL path.
+	 * @brief Add a new path resource with a callback
+	 * @param path URL path
 	 * @note Path should start with slash. Trailing slashes will be removed.
-	 * @param HttpPathDelegate callback - the callback that will handle this path
+	 * @param callback -The callback that will handle this path
 	 */
 	void addPath(String path, const HttpPathDelegate& callback);
+
 	void addPath(const String& path, const HttpResourceDelegate& onRequestComplete);
-	void addPath(const String& path, HttpResource* resource);
 
-	void setDefaultHandler(const HttpPathDelegate& callback);
-	void setDefaultResource(HttpResource* resource);
+	void addPath(const String& path, HttpResource* resource)
```

Comment:
Maybe deprecate some of these, use resourceTree directly?

Reply:
That is a good idea. Deprecate it now and the removal will be after two or three major releases. The Readme.md has to be updated too.

Reply:
I've renamed `resourceTree` to `paths` as it seems to make more sense to me when in use. I left the internal references (in `HttpServerConnection`) alone.